Assignment 2 Finisned by Lintong Li.Rmd

```diff
@@ -0,0 +1,196 @@
+EDGE1 <- select(EDGE, from, to)
+# create a new data frame to show the relation between stid and tid.
+names(EDGE1) <- c("tid", "stid")
+S.EDGE <- count(EDGE1, stid, stid)
+S.EDGE
+names(S.EDGE) <- c("id", "count")
+T.EDGE <- count(EDGE1, tid, tid)
+T.EDGE
+names(T.EDGE) <- c("id", "count")
+VERTEX2 <- bind_rows(T.EDGE, S.EDGE)
+VERTEX2
+VERTEX <- merge(VERTEX2, VERTEX, by = "id")
+VERTEX
+g2 = graph.data.frame(EDGE, directed = TRUE, vertices = VERTEX)
+g2
+plot(g2, layout = layout.fruchterman.reingold, vertex.size = VERTEX$count*3)
+```
+```
+
+
+
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+
+```
+```{r}
+library(igraph)
+library(dplyr)
+library(tidyr)
+# Getting the data frame from "hudk4050-classes.csv".
+D3 <- read.csv("hudk4050-classes.csv", header = TRUE)
+D3 <- as.data.frame(D3)
+D3 <- unite(D3, Name, "First.Name", "Last.Name", sep=" ")
+D3 <- filter(D3, Name != "ZIMO CHEN")
+# Getting the data frame which includes 3 colums(Name, Class.Name and count). 
+D4 <- gather(D3,Class.number, Class.Name, -Name)
+D4 <- filter(D4, Class.Name >0)
+D4 <- select(D4, Name,Class.Name)
+f1 <- function(a)
+{ gsub(" ", "", a)}
+D4 <- lapply(D4,f1)
+D4$count <- 1
+D4<- as.data.frame(D4)
+D4<-filter(D4, Class.Name!= "HUDK4050")
+#fixing the wrong information in D4. 
+D4$Class.Name <- gsub(pattern = "QMSS-",
+                                     replacement = "",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "QMSS",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "GG",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "GR",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+# Getting the Person-Course Matrix 
+D4 <- unique(D4)
+M1 <- spread(D4, Class.Name, count, fill=0)
+row.names(M1) <- M1$Name
+M1$Name <- NULL
+M1 <- as.matrix(M1)
+# Getting the Person to Person Matrix.
+M2 <- M1 %*% t(M1)
+diag(M2) <-NA
+#Graph 
+GraM2 <- graph.adjacency(M2, mode="undirected")
+plot.igraph(GraM2,layout=layout.fruchterman.reingold,vertex.size =15,vertex.label.cex =0.5)
+
+```
+
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+# Calculate the degree centrality of the nodes
+d <- degree(GraM2)
+d[d == max(d)]
+```
+```{r}
+# Calculate the betweeness centrality
+bc <- betweenness(GraM2)
+bc[bc == max(bc)]
```

Comment:
Overall, excellent work! Try to interpret the network metrics within the context of this dataset.